lib/rules/forbid-foreign-prop-types.js

```diff
@@ -121,7 +123,9 @@ module.exports = {
       },
 
       ObjectPattern(node) {
-        const propTypesNode = node.properties.find((property) => property.type === 'Property' && property.key.name === 'propTypes');
+        const propTypesNode = node.properties.find(
+          (property) => property.type === 'Property' && property.key.type === 'Identifier' && property.key.name === 'propTypes'
+        );
```

Comment:
this change doesn't seem needed

Reply:
Sorry🙇‍♂️  Fixed in https://github.com/jsx-eslint/eslint-plugin-react/commit/844d483855c56a6741cf56b0e3c9be2458d45e68 .